elyra/pipeline/validation.py

```diff
@@ -187,122 +183,107 @@ def _validate_pipeline_structure(self, pipeline: dict, response: ValidationRespo
         as well as verifying all nodes in the pipeline are supported by the runtime
         :param pipeline: the pipeline definition to be validated
-        :param pipeline_runtime: name of the pipeline runtime being used e.g. kfp, airflow, generic
-        :param pipeline_execution: name of the pipeline runtime for execution  e.g. kfp, airflow, local
+        :param pipeline_type: name of the pipeline runtime being used e.g. kfp, airflow, generic
+        :param pipeline_runtime: name of the pipeline runtime for execution  e.g. kfp, airflow, local
         :param response: ValidationResponse containing the issue list to be updated
         """
-        pipeline_json = json.loads(json.dumps(pipeline))
-        pipeline_id = pipeline_json['pipelines'][0]['id']
+
+        pipeline_id = pipeline_definition.primary_pipeline.id
         supported_ops = []
 
-        if pipeline_execution:
-            if pipeline_execution != pipeline_runtime and pipeline_runtime != 'generic':
+        if pipeline_runtime:
+            if pipeline_runtime != pipeline_type and pipeline_type != 'generic':
                 response.add_message(severity=ValidationSeverity.Error,
                                      message_type="invalidRuntime",
                                      message="Pipeline runtime platform is not compatible "
                                              "with selected runtime configuration.",
-                                     data={"pipelineRuntime": pipeline_runtime,
-                                           "pipelineID": pipeline_id})
-            elif pipeline_runtime == 'generic':
-                pipeline_runtime = 'local'  # Update to local since they both support the same set of ops
+                                     data={"pipelineID": pipeline_id,
+                                           "pipelineType": pipeline_type,
+                                           "pipelineRuntime": pipeline_runtime})
 
         if PipelineProcessorManager.instance().is_supported_runtime(pipeline_runtime):
             component_list = await PipelineProcessorManager.instance().get_components(pipeline_runtime)
             for component in component_list:
                 supported_ops.append(component.op)
 
             # Checks pipeline node types are compatible with the runtime selected
-            for single_pipeline in pipeline_json['pipelines']:
-                pipeline_id = single_pipeline['id']
-                node_list = single_pipeline['nodes']
-                for node in node_list:
-                    if node['type'] == "execution_node" and node['op'] not in supported_ops:
-                        node_label = node['app_data']['ui_data'].get('label', node['app_data']['label'])
+            for pipeline in pipeline_definition.pipelines:
+                for node in pipeline.nodes:
+                    if node.type == "execution_node" and node.op not in supported_ops:
```

Comment:
Why isn't this just part of Pipelinedfintion.Pipeline.Node.validate()?  I figured the PD would also validate since it has `validate()` methods at every level.

Reply:
this is not structure validation, but actually, check each op against the list of components registered to the runtime, so it's more like a validation piece. Would you agree?

Reply:
validation of the structure and validation of the values needs to occur.  I'm just asking why this code isn't more like: 
```python
for pipeline in pipeline_definition.pipelines:
    pipeline.validate()
```
and `pipeline.validate()` would look more like:
```python
    # validate pipeline data
    for node in self.nodes:
        node.validate()
```
and `node.validate() would look more like:
```python
    if self.type == "execution_node" and self.op not in supported_ops:
```